### PR TITLE
Use specified branches for ci updates

### LINKED ISSUE #684: Workflow for updating CI configs needs to only update a set of branches
Right now the workflow tries to update all branches going back to a certain one. The workflow should only update branches that are part of a list of supported branches.

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #684

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,8 +138,9 @@ The following settings are stored in `template_config.yml`.
                         'github'. To switch from Redmine to GitHub use the --migrate-github-issues
                         option.
 
+  ci_update_branches    Specify the branches that should receive regular CI updates.
   keep_ci_update_since_branch
-                        Keep CI updated for all starting from the specified branch.
+                        Keep CI updated for all starting from the specified branch. DEPRECATED
```
Comment (bmbouter) [inline README.md:143]: Can we just remove it? The worst case is plugin writers will not have some branches receive updates automatically, and then they would be like "oh I need to switch".

File: templates/github/.ci/scripts/update_ci_branches.py.j2 [hunk 2]
```diff
@@ -1,3 +1,5 @@
+{% include 'header.j2' %}
```
Comment (bmbouter) [inline templates/github/.ci/scripts/update_ci_branches.py.j2:1]: What does this do?

Comment (mdellweg) [reply templates/github/.ci/scripts/update_ci_branches.py.j2:1]: I tells you: "Can't touch this!"

File: README.md (hunk 1 above)
Comment (mdellweg) [reply README.md:143]: I'm trying to follow the principle of least surprise.

File: plugin-template [hunk 3]
```diff
@@ -257,6 +257,10 @@ def main():
                         if key not in config:
                             config[key] = value
                             write_new_config = True
+                    # remove deprecated options
+                    if "keep_ci_update_since_branch" in config:
```
Comment (bmbouter) [inline plugin-template:261]: Awww yeah!

**Pull request merged**